yamcs-core/src/main/java/org/yamcs/cascading/YamcsTcLink.java

```diff
@@ -63,15 +66,86 @@ public void init(String instance, String name, YConfiguration config) {
         } else {
             l = List.of(CommandHistoryPublisher.CcsdsSeq_KEY);
         }
+        if (config.containsKey("commandMapping")) {
+           List<YConfiguration> commandMapConfigList = config.getConfigList("commandMapping");
+           commandMapConfigList.forEach(conf -> commandMapDataList.add(new CommandMapData(conf)));
+        }
+
         keepUpstreamAcks = new HashSet<>(l);
     }
 
     public boolean sendCommand(PreparedCommand pc) {
-
         // TODO: map the command name based on some XTCE ancillary data
+        String pcCommand = pc.getMetaCommand().getQualifiedName();
+        String pcCommandPath = pcCommand.substring(0, pcCommand.lastIndexOf("/")+1);
+
+        for (CommandMapData data: commandMapDataList) {
+            String commandDataPath = data.GetLocalPath().substring(0, data.GetLocalPath().lastIndexOf("/")+1);
+
+            if (commandDataPath.equals(pcCommandPath) || data.GetLocalPath().equals(pcCommand)) {
+                switch (data.GetCommandType()) {
+                    case DIRECT:
+                        return sendDirectCommand(pc);
+                    case EMBEDDED_BINARY:
+                        return sendEmbeddedBinaryCommand(pc, data);
+                    default:
+                        log.warn("Wrong downstream command type.");
+                }
+            }
+        }
+        return true;
+    }
+
+    private boolean sendEmbeddedBinaryCommand(PreparedCommand pc, CommandMapData data) {
+        CommandInfo upstreamCmd = getUpstreamCmd(data.GetUpstreamPath());
```

Comment:
I don't see how you cover the case when the data.getUpstreamPath returns a system name (i.e. ends with "/") rather than a fully qualified command name. In that case you have to use the command name added to the system name.